docker/run.sh

```diff
@@ -1,24 +1,42 @@
 #! /usr/bin/env bash
 set -euo pipefail
 
+# Cd into directory holding this script
+cd "${BASH_SOURCE[0]%/*}"
+
 # Parameter controlling whether to attach GPUs to the Docker container
 ATTACH_GPUS=true
 
+# Amount of shared memory to give the Docker container access to
+# If you get a Bus Error, increase this value. If you don't have enough memory
+# on your machine, decrease this value.
+SHM_SIZE=8192m
+
 gpu_arg=""
 if $ATTACH_GPUS ; then gpu_arg="--gpus all" ; fi
-image=${1:-base}
+image=${1:-flexflow}
 
-# Cd into directory holding this script
-cd "${BASH_SOURCE[0]%/*}"
+FF_GPU_BACKEND=${FF_GPU_BACKEND:-cuda}
+if [[ "${FF_GPU_BACKEND}" != @(cuda|hip_cuda|hip_rocm|intel) ]]; then
+  echo "Error, value of FF_GPU_BACKEND (${FF_GPU_BACKEND}) is invalid. Pick between 'cuda', 'hip_cuda', 'hip_rocm' or 'intel'."
+  exit 1
+elif [[ "${FF_GPU_BACKEND}" != "cuda" ]]; then
+  echo "Configuring FlexFlow to build for gpu backend: ${FF_GPU_BACKEND}"
+else
+  echo "Letting FlexFlow build for a default GPU backend: cuda"
+fi
 
 
-if [[ "$image" == "base" ]]; then
-    eval docker run -it "$gpu_arg" flexflow:latest
+if [[ "$image" == "flexflow-environment" ]]; then
+    eval docker run -it "$gpu_arg" "--shm-size=${SHM_SIZE}" "flexflow-environment-${FF_GPU_BACKEND}:latest"
```

Comment:
Why `eval`?

Reply:
I think I added this to handle the `"$gpu_arg"` part, let me see if this works without the `eval` as well

Reply:
Ok, so removing the double quotes around `$gpu_arg` allow us to remove the `eval`, but we then get shellcheck errors:

```
Notice: ./docker/run.sh:31:20: note: Double quote to prevent globbing and word splitting. [SC2086]
Notice: ./docker/run.sh:33:20: note: Double quote to prevent globbing and word splitting. [SC2086]
Notice: ./docker/run.sh:36:20: note: Double quote to prevent globbing and word splitting. [SC2086]
```

I guess we could ignore the SC2086 error code for this file, but then we wouldn't catch other similar errors, so I'll just bring back the eval?

Reply:
Ah, didn't realize that included a flag. Yeah eval is fine